### PR TITLE
Update min/max/range folds to use slidingWindow

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

**Comment by harendra-kumar** [comment]
For benchmark build failures we need to remove gauge as dependency and use tasty-bench instead. tasty-bench has the same interface as gauge so can easily replace gauge. You can see the streamly-benchmarks.cabal and streamly benchmarks, they are also using tasty-bench now.

**Comment by harendra-kumar** [comment]
Can you squash the fixup commit?

File: src/Streamly/Statistics.hs [hunk 1]
```diff
@@ -52,74 +52,91 @@ data WindowSize
...
 {-# INLINE min #-}
-min :: Monad m => WindowSize -> Fold m Double Double
-min Infinite = Fold.foldl' step initial 
-  where
-    initial = 1 / 0
-    step ma a
-      | a < ma = a
-      | otherwise = ma
-min (Finite w) = Fold step initial extract
+min :: (Monad m, Num a, Ord a, Storable a) => Fold m (a, Maybe a) a
```
Comment (harendra-kumar) [inline src/Streamly/Statistics.hs:57]: Add the time complexity annotation as I suggested earlier.

File: src/Streamly/Statistics.hs [hunk 2]
```diff
@@ -52,74 +52,91 @@ data WindowSize
...
+      Just (ia', q') ->
+        if snd ia <= snd ia'
+          then dqloop ia q'
+          else DQ.snoc ia q
+
+  extract (Tuple3' _ _ q) = return $ snd
+                            $ fromMaybe (0, error "min: Empty stream")
+                            $ DQ.head q
+
+-- | The maximum element in the window.
```
Comment (harendra-kumar) [inline src/Streamly/Statistics.hs:95]: Add the time complexity annotation.

**Comment by specdrake** [comment]
@harendra-kumar can this be merged?

**Pull request merged**